crates/red_knot_module_resolver/src/resolver.rs

```diff
@@ -43,6 +43,43 @@ pub(crate) fn resolve_module_query<'db>(
     Some(module)
 }
 
+pub fn resolve_builtins(db: &dyn Db) -> Option<Module> {
```

Comment:
I would prefer if we don't use the resolve prefix for all salsa queries. I don't see it adding much

What's the reason for this wrapper function. It doesn't do anything

Reply:
> What's the reason for this wrapper function. It doesn't do anything

If you make this diff to my PR branch (which was what Carl and I tried initially):

```diff
--- a/crates/red_knot_module_resolver/src/db.rs
+++ b/crates/red_knot_module_resolver/src/db.rs
@@ -2,7 +2,7 @@ use ruff_db::Upcast;
 
 use crate::resolver::{
     editable_install_resolution_paths, file_to_module, internal::ModuleNameIngredient,
-    module_resolution_settings, resolve_builtins_query, resolve_module_query,
+    module_resolution_settings, resolve_builtins, resolve_module_query,
 };
 use crate::typeshed::parse_typeshed_versions;
 
@@ -12,7 +12,7 @@ pub struct Jar(
     module_resolution_settings,
     editable_install_resolution_paths,
     resolve_module_query,
-    resolve_builtins_query,
+    resolve_builtins,
     file_to_module,
     parse_typeshed_versions,
 );
diff --git a/crates/red_knot_module_resolver/src/resolver.rs b/crates/red_knot_module_resolver/src/resolver.rs
index 28c7f0222..a4945054e 100644
--- a/crates/red_knot_module_resolver/src/resolver.rs
+++ b/crates/red_knot_module_resolver/src/resolver.rs
@@ -43,12 +43,8 @@ pub(crate) fn resolve_module_query<'db>(
     Some(module)
 }
 
-pub fn resolve_builtins(db: &dyn Db) -> Option<Module> {
-    resolve_builtins_query(db)
-}
-
 #[salsa::tracked]
-pub(crate) fn resolve_builtins_query(db: &dyn Db) -> Option<Module> {
+pub fn resolve_builtins(db: &dyn Db) -> Option<Module> {
     let _span = tracing::trace_span!("resolve_builtins").entered();
```

then Clippy issues this complaint, despite the fact that it's publicly exported from the crate in `lib.rs`:

```
warning: unreachable `pub` item
  --> crates/red_knot_module_resolver/src/resolver.rs:47:1
   |
47 | pub fn resolve_builtins(db: &dyn Db) -> Option<Module> {
   | ---^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   | |
   | help: consider restricting its visibility: `pub(crate)`
   |
   = help: or consider exporting it for use by other crates
   = note: requested on the command line with `-W unreachable-pub`
```

We found that using a wrapper function solved this issue, and I suppose we assumed that this was the way to go since this file already has `resolve_module()`, which is just a thin wrapper around `resolve_module_query()`. But perhaps it's better just to suppress the bogus Clippy warning in this instance?

Reply:
Renaming to `builtins_module` instead of `resolve_builtins`.

The reason for the wrapped function is that there seems to be an issue with making a salsa query `pub` -- even if we re-export it in `lib.rs` it still issues an "unreachable pub item" warning. I think this is maybe because `#[salsa::tracked]` creates both a function and a struct by that name? Not sure.

Anyway, I can just allow that warning instead of using the no-op wrapper function.

Reply:
Yeah. That's a clippy issue that we already suppress in other places and is something we have to fix upstream in salsa